src/wccp2.cc

```diff
@@ -1452,7 +1452,7 @@ wccp2HandleUdp(int sock, void *)
         ptr += sizeof(*routerCountRaw);
         const auto ipCount = ntohl(*routerCountRaw);
         const auto ipsSize = ipCount * sizeof(struct in_addr); // we check for unsigned overflow below
-        Must3(ipsSize / sizeof(struct in_addr) != ipCount, "huge IP address count", Here());
+        Must3(ipsSize / sizeof(struct in_addr) == ipCount, "huge IP address count", Here());
```

Comment:
Thank you for fixing this bug. The condition description is still wrong/inverted, but all other Must3() calls in wccp2.cc suffer the same flaw, so it is probably best to leave that problem unfixed in _this_ PR.